traffic_ops/app/lib/API/Cdn.pm

```diff
@@ -123,10 +126,14 @@ sub create {
 		return $self->alert( "a cdn with domain " . $params->{domainName} . " already exists." );
 	}
 
+	#setting tenant_id to the user's tenant if tenant is not set. TODO(nirs): remove when tenancy is no longer optional in the API
+	my $tenant_id = exists($params->{tenantId}) ? $params->{tenantId} :  $self->current_user_tenant();
+
 	my $values = {
 		name => $params->{name},
 		dnssec_enabled => $params->{dnssecEnabled},
 		domain_name => $params->{domainName},
+		tenant_id => $tenant_id,
```

Comment:
oh, nevermind, i see what you are doing here. how about this instead?

tenant_id => $params->{tenantId} // $self->current_user_tenant();

meaning, set tenant_id to $params->{tenantId} if it's not null / undef, otherwise set tenant_id to $self->current_user_tenant()